### PR TITLE
Testing refactor

### LINKED ISSUE #1: Pantry Class
Every User should have a pantry. A Pantry holds on to all the ingredients its owner has stocked, and the amount of each ingredient they have. As a user, I should be able to:

- [ ] A method to determine whether my pantry has enough ingredients to cook a given meal
- [ ] A method to determine the amount of ingredients still needed to cook a given meal, based on what’s in my pantry
- [ ] A method to remove the ingredients used for a given meal from my pantry, once that meal has been cooked (only applicable if users have a list of mealsToCook; can be considered a stretch goal)
- [ ] Determine how much it will cost to buy the necessary ingredients needed to cook a given meal, based on what’s in my pantry
- [ ] Add the necessary ingredients to my pantry (and keep this up to date with the database via fetch)

### LINKED ISSUE #9: Create a test file for Pantry class
- [ ] Initial values of class properties need tests
- [ ] Class methods need tests for all expected outcomes
- [ ] Any methods that modify class properties should be test
Note - You are not required to test your fetch calls

### PR DESCRIPTION
**Pull request by KJJones13**
#### What’s this PR do?  
 Refactors the testing and adds a few necessary methods to the classes. It also adds some sad path test.

#### Where should the reviewer start?  
 All of the original classes and test files

#### How should this be manually tested?  
 run npm test

#### Any background context you want to provide?  
 I tried to control the incoming data while providing sad path test

#### What are the relevant tickets?  
 closes #9, closes #1
 
#### Questions: 
There are probably some more sad path testing that we could add? Let me know if y'all see any other opportunities. It looks like there will be some merge conflicts like I mentioned in our conversation.

### CODE REVIEW

File: src/user.js [hunk 1]
```diff
@@ -7,8 +7,23 @@ class User {
...
+    switch (dataType) {
+      case 'string':
+        return 'Invalid value given';
+        break;
+      case 'number':
+        return 0;
+        break;
+    }
+  }
+
```
Comment (BlakeDonn) [inline src/user.js:24]: This is really cool, It pretty much just evaluates the typof dataType and changes return value based on if it is a string or number?

File: src/user.js [hunk 2]
```diff
@@ -33,12 +48,30 @@ class User {
...
+    this.recipesToCook.splice(cooked, 1)
+  }
+
+  filterRecipesToCook(tag) {
+    return this.recipesToCook.filter(recipe => {
+      return recipe.tags.includes(tag);
+    })
+  }
+
+
```
Comment (BlakeDonn) [inline src/user.js:68]: Looks good to me, need to consider dom functionality for this. Probably something similar to the way the displaying of recipe directions is handled could work.

**Review by BlakeDonn — APPROVED** [review]
Thanks for making the test files more robust, and handling sad-path functionality for the data so we can avoid pushing bad data.

File: src/user.js (hunk 1 above)
Comment (Atos20) [reply src/user.js:24]: Great use of the switch statement, I know how to read it but have never used it before, I will give it a try

**Comment by Atos20** [comment]
We can resolve the merge conflict at 5 when we get back together!

File: src/user.js (hunk 1 above)
Comment (Atos20) [reply src/user.js:24]: I haven't tried using this switch statement, they look very straight forward and will try to implement some of the moving forward

File: src/user.js [hunk 3]
```diff
@@ -33,12 +48,30 @@ class User {
...
   static getUserData(userId) {
     const userUrl = 'https://fe-apps.herokuapp.com/api/v1/whats-cookin/1911/users/wcUsersData'
     const promise = fetch(userUrl)
       .then(response => response.json())
       .then(data => data.wcUsersData.find(user => user.id === userId))
-    return promise; 
+    return promise;
```
Comment (Atos20) [inline src/user.js:74]: This is something we talked before but just want to leave it here so we don't forget and that is implementing error handling.

File: test/cookbook-test.js [hunk 4]
```diff
@@ -6,22 +6,28 @@ import Cookbook from '../src/cookbook';
...
-      expect(cookbook.findRecipe('yolk').length).to.equal(2);
-    });
+  it('Should be able to filter through its array by ingredients', () => {
+    expect(cookbook.findRecipe('yolk').length).to.equal(2);
+  });
 
-    it('Should be able to filter through its array by name', () => {
-      expect(cookbook.findRecipe('Sesame Cookies').length).to.equal(1);
-    });
+  it('Should be able to filter through its array by name', () => {
```
Comment (Atos20) [inline test/cookbook-test.js:30]: Thanks for adding the missing testing for the Cookbook class, I think we can 2 or 3 sad paths for this specific class

File: test/pantry-test.js [hunk 5]
```diff
@@ -44,7 +54,7 @@ describe('Pantry', () => {
       { id: 1012047, amountNeeded: 24 },
       { id: 10019903, amountNeeded: 2 },
       { id: 2050, amountNeeded: 0.5 }
-    ])
+    ]);
```
Comment (Atos20) [inline test/pantry-test.js]: thanks also for adding missing semicolons

File: test/user-test.js [hunk 6]
```diff
@@ -54,6 +80,32 @@ describe('User', () => {
...
+    expect(user1.recipesToCook).to.eql([]);
+  });
+
+  it('Should be able to add recipes to recipesToCook', () => {
+    user1.addToCook(recipeData[0]);
+    user1.addToCook(recipeData[1]);
+    expect(user1.recipesToCook.includes(recipeData[0])).to.equal(true);
+  });
+
+  it('Should not add recipes to recipesToCook if it is undefined', () => {
```
Comment (Atos20) [inline test/user-test.js:94]: good catch for all these sad paths!!

**Review by Atos20 — APPROVED** [review]

**Comment by Atos20** [comment]
Merge conflicts were resolved and they were mostly testing I had implemented before that weren't in the old version

**Pull request merged**